src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs

```diff
@@ -18,106 +19,60 @@
 using Voron;
 using Voron.Data.Tables;
 using Voron.Impl;
+using Raven.Server.ServerWide.Commands.Subscriptions;
+using Raven.Client.Documents.Replication.Messages;
+using System.Threading.Tasks;
+using Raven.Client.Documents;
+using Raven.Client.Json.Converters;
+using Raven.Server.Rachis;
 
 namespace Raven.Server.Documents.Subscriptions
 {
     // todo: implement functionality for limiting amount of opened subscriptions
     public class SubscriptionStorage : IDisposable
     {
         private readonly DocumentDatabase _db;
+        private readonly ServerStore _serverStore;
         public static TimeSpan TwoMinutesTimespan = TimeSpan.FromMinutes(2);
         private readonly ConcurrentDictionary<long, SubscriptionState> _subscriptionStates = new ConcurrentDictionary<long, SubscriptionState>();
-        private readonly TableSchema _subscriptionsSchema = new TableSchema();
-        private readonly StorageEnvironment _environment;
         private readonly Logger _logger;
 
-
-        public SubscriptionStorage(DocumentDatabase db)
+        public SubscriptionStorage(DocumentDatabase db, ServerStore serverStore)
         {
             _db = db;
-            var path = db.Configuration.Core.DataDirectory.Combine("Subscriptions");
-
-            var options = db.Configuration.Core.RunInMemory
-                ? StorageEnvironmentOptions.CreateMemoryOnly(path.FullPath, null, db.IoChanges, db.CatastrophicFailureNotification)
-                : StorageEnvironmentOptions.ForPath(path.FullPath, null, null, db.IoChanges, db.CatastrophicFailureNotification);
-
-            options.OnNonDurableFileSystemError += db.HandleNonDurableFileSystemError;
-            options.OnRecoveryError += db.HandleOnRecoveryError;
-
-            options.SchemaVersion = 1;
-            options.TransactionsMode = TransactionsMode.Lazy;
-            options.ForceUsing32BitsPager = db.Configuration.Storage.ForceUsing32BitsPager;
-            options.TimeToSyncAfterFlashInSeconds = db.Configuration.Storage.TimeToSyncAfterFlashInSeconds;
-            options.NumOfCocurrentSyncsPerPhysDrive = db.Configuration.Storage.NumOfCocurrentSyncsPerPhysDrive;
-            options.MasterKey = db.MasterKey;
-
-
-            _environment = new StorageEnvironment(options);
-            var databaseName = db.Name;
-
-            _logger = LoggingSource.Instance.GetLogger<SubscriptionStorage>(databaseName);
-            _subscriptionsSchema.DefineKey(new TableSchema.SchemaIndexDef
-            {
-                StartIndex = 0,
-                Count = 1
-            });
+            _serverStore = serverStore;
+            _logger = LoggingSource.Instance.GetLogger<SubscriptionStorage>(db.Name);
         }
 
         public void Dispose()
         {
             foreach (var state in _subscriptionStates.Values)
             {
                 state.Dispose();
```

Comment:
There is no urgent need right now(we have a CTS.Cancel in there), but I'll add it anyway, in case that code inside the dispose will change one day..